### PR TITLE
feat(primitives): add avatar primitive

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adding Avatar primitive to be use in the AIConversation component. 

```jsx
{/* Avatar with image */}
<Avatar src="/cats/5.jpg" />
{/* Avatar with default placeholder icon */}
<Avatar />
{/* Avatar with initials */}
<Avatar>DB</Avatar>
{/* Avatar with custom icon */}
<Avatar>
  <FiSmile style={{ width: '60%', height: '60%' }} />
</Avatar>
```

#### Description of how you validated changes

CleanShot 2024-08-20 at 15 23 35

CleanShot 2024-08-20 at 14 19 05@2x

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [x] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/avatar/examples/DefaultAvatarExample.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      {/* Avatar with image */}
+      <Avatar src="/cats/5.jpg" />
+      {/* Avatar with default placeholder icon */}
+      <Avatar />
+      {/* Avatar with initials */}
+      <Avatar>DB</Avatar>
+      {/* Avatar with custom icon */}
+      <Avatar>
+        <FiSmile style={{ width: '60%', height: '60%' }} />
+      </Avatar>
```
Comment (thaddmt) [inline docs/src/pages/[platform]/components/avatar/examples/DefaultAvatarExample.tsx:16]: Just clarifying that for our use case with AIConversation we will just pass two separate instances of this `Avatar` with a custom icon for the user and the assistance right?

**Review by thaddmt — DISMISSED** [review]

File: docs/src/pages/[platform]/components/avatar/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    (src ? `\n  src=${JSON.stringify(src)}` : '') +
+    (size ? `\n  size=${JSON.stringify(size)}` : '') +
+    (colorTheme ? `\n  colorTheme=${JSON.stringify(colorTheme)}` : '') +
+    (variation ? `\n  variation=${JSON.stringify(variation)}` : '') +
+    `\n/>`
+  );
+};
+
+const defaultAvatarProps = {
+  src: '/cats/1.jpg',
```
Comment (hbuchel) [inline docs/src/pages/[platform]/components/avatar/demo.tsx:22]: Just a nit: should we use a transparent avatar or svg icon so the demo can show the variations better? Currently the demo doesn't make it clear that filled/default variations do much when you change them.

File: docs/src/pages/[platform]/components/avatar/examples/DefaultAvatarExample.tsx (hunk 1 above)
Comment (dbanksdesign) [reply docs/src/pages/[platform]/components/avatar/examples/DefaultAvatarExample.tsx:16]: Yup

**Review by hbuchel — COMMENTED** [review]
LGTM! 

Can we include an accessibility section with an example of using alt text with an image? It wasn't immediately clear to me that I could pass an alt attribute to `<Avatar />` and it would find it's way to the ``

The default placeholder Avatar is essentially non-existent to screen readers. A developer could overcome this by passing an `aria-label` to `<Avatar />` with whatever placeholder text makes sense for their application. I don't think that needs to be "braked in" but maybe we could show that in a11y example, too?

**Review by hbuchel — APPROVED** [review]

**Review by thaddmt — APPROVED** [review]

**Pull request merged**